### DISCUSSION TITLE
Backend Sessions

### DISCUSSION THREAD

**Discussion by NgLoader** [highlight/highlight#5029] [Ideas]
Incorporating a backend session replay feature into web applications can significantly improve the backend experience, similar to how clients benefit from session replay.
There are several compelling reasons why this feature should be considered:

1. Customized Hardware: In many cases, the backend of a web application is hosted on the customer's servers, which may have custom hardware configurations and background tasks.
By implementing a backend session replay feature, developers can gain insights into how the application is performing on this unique hardware, enabling them to better support customers.

2. Multiple Servers: Another common scenario is when the backend is distributed across multiple servers, which makes it challenging to analyze logs and identify issues.
A backend session replay feature would allow developers to replay sessions across all servers and gain a view of how the application is behaving, thereby making it easier to troubleshoot and optimize performance.

3. Metrics: Just getting some more details of how the currently backend is running like ``cpu usage``, ``api request``, ``ram usage`` and so one.

It could also be separated into the frontend with two sections ``frontend`` and ``backend`` under ``sessions``.

I understand when the Highlight team may not currently plan to support this feature, as it could change the nature of the project.
I hope to get some feedback from the community what they think about this.

Issue: https://github.com/highlight/highlight/issues/4446

**Comment by jay-khatri** [comment]
Hey @NgLoader , this is very interesting. I wonder, though, if this overlaps with features that we could include in a future-planned tracing product? See:

https://github.com/highlight/highlight/issues/4446

**Reply by NgLoader** [reply]
Yea, thats what I was thinking about
But I also suggesting a session/login implementation for backend sdks for more security and only valid authentication access

**Reply by jay-khatri** [reply]
Understood. This discussion will become much more relevant when we decide to go into tracing (likely in Q3). I'll close this discussion until then.

**Reply by NgLoader** [reply]
Okay
